src/java/htsjdk/samtools/BAMIndexer.java

```diff
@@ -55,6 +55,10 @@
      */
     public BAMIndexer(final File output, final SAMFileHeader fileHeader) {
 
+        if (fileHeader.getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
```

Comment:
It's possible we should allow "unsorted" to pass with a warning.  There may be bams that don't have the sort order set that we want to index anyway assuming they're coordinate sorted.